queries/clojure/highlights.scm

```diff
@@ -1,77 +1,333 @@
+  "print" "print-ctor" "print-dup" "print-method" "print-simple"
+  "print-str" "printf" "println" "println-str" "prn" "prn-str"
+  "promise" "proxy-call-with-super" "proxy-mappings" "proxy-name"
+  "push-thread-bindings" "qualified-ident?" "qualified-keyword?"
+  "qualified-symbol?" "quot" "rand" "rand-int" "rand-nth" "random-sample"
+  "range" "ratio?" "rational?" "rationalize" "re-find" "re-groups"
+  "re-matcher" "re-matches" "re-pattern" "re-seq" "read"
+  "read+string" "read-line" "read-string" "reader-conditional"
+  "reader-conditional?" "realized?" "record?" "reduce"
+  "reduce-kv" "reduced" "reduced?" "reductions" "ref" "ref-history-count"
+  "ref-max-history" "ref-min-history" "ref-set" "refer"
+  "release-pending-sends" "rem" "remove" "remove-all-methods"
+  "remove-method" "remove-ns" "remove-tap" "remove-watch"
+  "repeat" "repeatedly" "replace" "replicate"
+  "requiring-resolve" "reset!" "reset-meta!" "reset-vals!"
+  "resolve" "rest" "restart-agent" "resultset-seq" "reverse"
+  "reversible?" "rseq" "rsubseq" "run!" "satisfies?"
+  "second" "select-keys" "send" "send-off" "send-via"
+  "seq" "seq?" "seqable?" "seque" "sequence" "sequential?"
+  "set" "set-agent-send-executor!" "set-agent-send-off-executor!"
+  "set-error-handler!" "set-error-mode!" "set-validator!"
+  "set?" "short" "short-array" "shorts" "shuffle"
+  "shutdown-agents" "simple-ident?" "simple-keyword?"
+  "simple-symbol?" "slurp" "some" "some-fn" "some?"
+  "sort" "sort-by" "sorted-map" "sorted-map-by"
+  "sorted-set" "sorted-set-by" "sorted?" "special-symbol?"
+  "spit" "split-at" "split-with" "str" "string?"
+  "struct" "struct-map" "subs" "subseq" "subvec" "supers"
+  "swap!" "swap-vals!" "symbol" "symbol?" "tagged-literal"
+  "tagged-literal?" "take" "take-last" "take-nth" "take-while"
+  "tap>" "test" "the-ns" "thread-bound?" "to-array"
+  "to-array-2d" "trampoline" "transduce" "transient"
+  "tree-seq" "true?" "type" "unchecked-add" "unchecked-add-int"
+  "unchecked-byte" "unchecked-char" "unchecked-dec"
+  "unchecked-dec-int" "unchecked-divide-int" "unchecked-double"
+  "unchecked-float" "unchecked-inc" "unchecked-inc-int"
+  "unchecked-int" "unchecked-long" "unchecked-multiply"
+  "unchecked-multiply-int" "unchecked-negate" "unchecked-negate-int"
+  "unchecked-remainder-int" "unchecked-short" "unchecked-subtract"
+  "unchecked-subtract-int" "underive" "unquote"
+  "unquote-splicing" "unreduced" "unsigned-bit-shift-right"
+  "update" "update-in" "update-proxy" "uri?" "uuid?"
+  "val" "vals" "var-get" "var-set" "var?" "vary-meta" "vec"
+  "vector" "vector-of" "vector?" "volatile!" "volatile?"
+  "vreset!" "with-bindings*" "with-meta" "with-redefs-fn" "xml-seq"
+  "zero?" "zipmap"))
+
+
+
+;; >> Context based highlighting
 
-;; other calls
+; def-likes
+; Correctly highlight docstrings
 (list_lit
  .
- (sym_lit) @function)
+ (sym_lit) @a ; Don't really want to highlight twice
+ (#match? @a "^def.*")
+ .
+ (sym_lit)
+ .
+ (str_lit)? @text
```

Comment:
Text?

Reply:
It's a docstring.
I think it should be highlighted differently to normal strings but I wasn't sure how, maybe `@comment` would work better 🤔

Reply:
I tried changing this to `@comment` but I start getting wierd bugs.
Like when I open the buffer the docstring is highlighted as a comment, but if I select it then it get's highlighted as a string when I unselect.
And `:TSHighlightCapturesUnderCursor` just shows it as a string, very strange 🤔.

I'm going to remove the capture and add a TODO comment above to re-add it.

Reply:
Docstrings should make use of injections to highlight them as the markup they use, like markdown in rust or rst in python.

Reply:
Ah, that makes sense.

Unfortunately changing it to `@comment` seems to be a bit buggy atm.

Reply:
You need to include this in the `injections.scm` file, about the double hl, that's probably because of the order of the queries, the queries below others have more priority.

Reply:
I'm happy to add the docstrings to `injections.scm`, that's a great idea.

As for the double hl, I think it's more a bug about when the highlights get re-applied after buffer changes, here's a gif of the issue I'm having:
<img src="https://user-images.githubusercontent.com/8889986/123473176-1cfad500-d5f0-11eb-9ac8-56d6e6c2c805.gif" width=400>

In the above I type roughly this: `rs:w<cr>:e<cr>`

It's on the `rs` that the docstring changes from an `@comment` to an `@string`.
And when I type `:e<cr>` it changes back.

You can also get some strange things when editing multi-line docstrings:

Typing `gg` breaks the docstring highlight:
<img src="https://user-images.githubusercontent.com/8889986/123473869-1751bf00-d5f1-11eb-8fde-0b69be2d9de6.gif" width=400>

Having the docstring page off screen breaks the highlight: 
<img src="https://user-images.githubusercontent.com/8889986/123474338-b8d91080-d5f1-11eb-96ea-4eda773423ff.gif" width=400>
But only when you are in insert mode inserting newlines, the `o` command works fine for some reason 🤷‍♂️